tests/foreman/api/test_permission.py

```diff
@@ -5,34 +5,56 @@
 tested can be found here: http://theforeman.org/api/apidoc/v2/permissions.html
 
 """
+import json
+import random
 import re
-from ddt import data as ddt_data, ddt
+
+from ddt import ddt
 from fauxfactory import gen_alphanumeric
 from itertools import chain
 from nailgun import client, entities
 from requests.exceptions import HTTPError
 from robottelo.common.constants import PERMISSIONS
 from robottelo.common.decorators import bz_bug_is_open, data, run_only_on
-from robottelo.common.helpers import get_nailgun_config, get_server_credentials
+from robottelo.common.helpers import (
+    get_nailgun_config,
+    get_server_credentials,
+    get_server_software,
+)
 from robottelo.test import APITestCase
 # (too-many-public-methods) pylint:disable=R0904
 
 
-#: e.g. ['Architecture', 'Audit', 'AuthSourceLdap', …]
-PERMISSION_RESOURCE_TYPES = [key for key in PERMISSIONS.keys()]
-#: e.g. ['view_architectures', 'create_architectures', 'edit_architectures', …]
-PERMISSION_NAMES = [
-    value for value in chain.from_iterable(PERMISSIONS.values())
-]
-
-
 @ddt
 class PermissionsTestCase(APITestCase):
     """Tests for the ``permissions`` path."""
+    @classmethod
+    def setUpClass(cls):
+        super(PermissionsTestCase, cls).setUpClass()
+        cls.permissions = PERMISSIONS.copy()
+        if get_server_software() == 'upstream':
+            cls.permissions[None] += cls.permissions.pop('DiscoveryRule')
+            permissions = list(cls.permissions[None])
+            permissions.remove('app_root')
+            permissions.remove('attachments')
+            permissions.remove('configuration')
+            permissions.remove('logs')
+            permissions.remove('view_cases')
+            permissions.remove('view_log_viewer')
+            permissions.remove('view_search')
+            cls.permissions[None] = tuple(permissions)
+
+        #: e.g. ['Architecture', 'Audit', 'AuthSourceLdap', …]
+        cls.permission_resource_types = [
+            key for key in cls.permissions.keys()
+        ]
+        #: e.g. ['view_architectures', 'create_architectures', …]
+        cls.permission_names = [
+            value for value in chain.from_iterable(cls.permissions.values())
+        ]
 
     @run_only_on('sat')
-    @ddt_data(*PERMISSION_NAMES)
```

Comment:
This will decrease the number of tests, and we can't check for server software in the import time.